lib/net.js

```diff
@@ -933,12 +949,16 @@ Socket.prototype.connect = function(options, cb) {
 
 
 Socket.prototype.ref = function() {
+  this._unref = false;
+
   if (this._handle)
     this._handle.ref();
 };
 
 
 Socket.prototype.unref = function() {
+  this._unref = new Date();
```

Comment:
I had originally used a date after looking at https://github.com/joyent/node/pull/9115, but I went ahead and changed it to just using a boolean